runtime/src/chplmemtrack.c

```diff
@@ -173,8 +173,9 @@ void chpl_setMemFlags(void) {
     if (chpl_numNodes == 1) {
       memLogFile = fopen(memLog, "w");
     } else {
-      char* filename = (char*)sys_malloc((strlen(memLog)+10)*sizeof(char));
-      sprintf(filename, "%s.%" PRI_c_nodeid_t, memLog, chpl_nodeID);
+      int size = (strlen(memLog)+10)*sizeof(char);
+      char* filename = (char*)sys_malloc(size);
+      snprintf(filename, size, "%s.%" PRI_c_nodeid_t, memLog, chpl_nodeID);
```

Comment:
here, although we multiply by `sizeof(char)` when declaring `size` it it declared as an `int`